### ISSUE TITLE
Empty Sequential has different semantics than other cases

### ISSUE THREAD

**Issue by niboshi** [chainer/chainer#6303]
As you can see from below, an empty `chainer.Sequential()` has different semantics: it wraps an argument with a tuple and return it. If at least one built-in links or functions exist, it returns a single `Variable` object.

I think it's better to forbid calling an empty `Sequential`.

#### Code
```py
import numpy
import chainer
from chainer import functions as F

x = numpy.array([1, 2], numpy.float32)
seq = chainer.Sequential()
ret = seq(x)

print(ret)
print(type(ret))
print(type(ret[0]))
print()

seq = chainer.Sequential(F.tanh)
ret = seq(x)
print(type(ret))
```
#### Result
```
(array([1., 2.], dtype=float32),)
<class 'tuple'>
<class 'numpy.ndarray'>

<class 'chainer.variable.Variable'>
```

**Comment by toslunar** [comment]
I don't think it's "different".  Chainer functions may return `Variable`, `ndarray`, or tuple of them.  Example: `F.max_pooling_2d(return_indices=True)`.

**Comment by niboshi** [comment]
@toslunar 
That's because `F.max_pooling_2d(return_indices=True)` itself returns such values. That's a different story, I think.

Let me rephrase:
`Sequential` can be considered as a sequence of transforms. If there are one or more operations, that holds. If there's no operation, one would expect the input is returned as is. But currently this is not the case in empty `Sequential`.

**Comment by toslunar** [comment]
What do you think `chainer.Sequential()(x0, x1)` should return?

**Comment by niboshi** [comment]
@toslunar 
I think there's no reasonable result everyone can agree with.
That's why I want to forbid calling an empty `Sequential`.

**Comment by toslunar** [comment]
Therefore I don't think the semantics is "different".
